synapse/util/metrics.py

```diff
@@ -64,6 +64,12 @@
     sub_metrics=["real_time_max", "real_time_sum"],
 )
 
+# This is dynamically created in InFlightGauge.__init__.
```

Comment:
I seem to remember @squahtx  had a plan for this somewhere.

Looks like PR https://github.com/matrix-org/synapse/pull/10847 . Can we revisit this?

Reply:
If #10847 is finished up then sure!

Reply:
Looks like I ended up doing practically the same thing as this PR. I wouldn't block either PR on the other.